### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by lukicdarkoo**
**Description**
Refactors documentation

**Related Issues**
#147

**Affected Packages**
  - webots_ros2

**Tasks**
  - [x] Move all documentation from ROS wiki
  - [ ] Update ROS wiki with a simple introduction and link to this documentation
  - [x] Add table of contents
  - [x] Add tutorials (maybe in another PR)
  - [x] Refactor documentation in `webots_ros2_core`

### CODE REVIEW

**Comment by lukicdarkoo** [comment]
@cyberbotics/maintainers, I would like that we create a solid base for documentation, so users do not get confused. Currently, we don't have an overview of `webots_ros2` documentation, and the documentation scattered around a few places (ROS Wiki, this repo, inline documentation) without clear rules. Please feel free to edit this post and comment!

### Documentation distribution
According to the discussion, this is a plan:
- ROS Wiki: Just a quick introduction to Webots and `webots_ros2`, and visible link to this repository.
- This repo: Each package has to contain a README with 2-3 sentences describing the purpose of the package with pointers to GitHub wiki.
- GitHub Wiki: Actual place at which documentation is stored.
- Documentation should be shared for all ROS2 distributions and ROS2 distribution differences should be explicitly highlighted.

### Table of contents

- Getting Started
- Build and Install
- Examples
  - Wheeled robot
    - ...
  - Robotic arms
    - ...
  - *Vehicles
    - ...
  - *Drones
    - ...
- Tutorials
  - **Custom robot in Webots
  - Custom ROS2 driver
  - *Running Webots and ROS2 in CI
  - Navigation with TurtleBot3
  - SLAM with TurtleBot3
  - *MoveIt with Universal Robots
  - *Autoware with Tesla 3
- Guide
  - Webots Launcher
  - Webots Node
    - Webots Differential Drive Node
    - Webots Robotic Arm Node
  - Joint State Publisher
  - Devices
    - Camera
    - IMU
    - ...
  - Utils

\* We don't have yet that documentation
** External link

**Comment by lukicdarkoo** [comment]
My concern here is inline documentation, e.g.:
https://github.com/cyberbotics/webots_ros2/blob/662ba796fa3bd26f548a0d376e7915c823028163/webots_ros2_core/webots_ros2_core/devices/camera_device.py#L24
we should have a mechanism to publish this documentation as well.
The benefit of GitHub Wiki is that we can iterate fast (which is useful at this point) while creating markdown files that can be reused later. However, we cannot publish inline documentation.

**Comment by DavidMansolino** [comment]
The table of content looks very good.

> we should have a mechanism to publish this documentation as well.

There are probably some python modules to generate documentation for such formatted comments.

> The benefit of GitHub Wiki is that we can iterate fast (which is useful at this point) while creating markdown files that can be reused later. However, we cannot publish inline documentation.

Why not using markdown files and publish them on the gh-pages branches, wiki is indeed fast, but I always have the feeling that we have only poor control on it (hard to point to a specific tag/release and no review/notification for changes)

**Comment by omichel** [comment]
The table of contents looks very good to me as well.
I agree that the device pages (Camera, IMU, etc.), or at least some device class description pages, could be automatically generated by the CI from the source code comments.
Adding another layer transforming the MD files to HTML on the gh-pages branch has indeed some advantages but also some disadvantages, like it's more obvious for newbies to start looking into the wiki pages. Also opening up these wiki pages for edition by users should encourage easy contributions. So I wouldn't implement this now, but maybe at a later stage if we believe the wiki is not satisfactory.

**Comment by DavidMansolino** [comment]
> Adding another layer transforming the MD files to HTML on the gh-pages branch has indeed some advantages but also some disadvantages, like it's more obvious for newbies to start looking into the wiki pages.

This is not really required, gh-pages takes as input the MD files directly, there is no need for us to commit any HTML files.

**Comment by DavidMansolino** [comment]
👀

File: .github/workflows/documentation.yml
Comment (DavidMansolino) [inline .github/workflows/documentation.yml:7]: Don't forget to remove this before merging.

**Review by DavidMansolino — CHANGES_REQUESTED** [review]
That looks very good and complete! I didn't check completely the wiki yet, it would be nice if we can somehow display in the page when it was auto-generated (so that we don't edit it).

File: webots_ros2_epuck/README.md
Comment (DavidMansolino) [inline webots_ros2_epuck/README.md:3]:
```suggestion
This package provides a ROS2 interface example of the simulated [e-puck](https://www.gctronic.com/doc/index.php/e-puck2) robot in Webots.
```

File: webots_ros2_tiago/README.md
Comment (DavidMansolino) [inline webots_ros2_tiago/README.md:3]:
```suggestion
This package provides a ROS2 interface example for the simulated TIAGo robot in Webots.
```

File: .github/workflows/documentation.yml
Comment (lukicdarkoo) [reply .github/workflows/documentation.yml:32]: `${GITHUB_ACTOR}` is a username, not a name, and `${GITHUB_ACTOR}@github.com` is not associated with any GitHub account. But maybe, this isn't important because the commits are hidden in a way.

File: .github/workflows/documentation.yml
Comment (lukicdarkoo) [reply .github/workflows/documentation.yml]: It seems to be working without specifying in `env`.

**Comment by lukicdarkoo** [comment]
@DavidMansolino @omichel Does the main README.md look appealing? User engagement depends a lot on the main README.md, so if you have any ideas let me know :)

**Comment by DavidMansolino** [comment]
> @DavidMansolino @omichel Does the main README.md look appealing? User engagement depends a lot on the main README.md, so if you have any ideas let me know :)

Yes, I like the image!

File: README.md
Comment (omichel) [inline README.md:13]:
```suggestion
Please visit the [documentation](https://github.com/cyberbotics/webots_ros2/wiki) that contains the following sections:
```

**Review by omichel — CHANGES_REQUESTED** [review]
The README.md looks good to me, it is simple and attractive with a nice image and a short list of appealing sections. 👍

**Comment by omichel** [comment]
I believe that the Wiki pages which are automatically generated should include a header explaining this, e.g., "Do not edit these wiki pages as they are automatically generated from the Python docstrings. If you edit them, your changes will be lost.".

**Comment by lukicdarkoo** [comment]
> I believe that the Wiki pages which are automatically generated should include a header explaining this, e.g., "Do not edit these wiki pages as they are automatically generated from the Python docstrings. If you edit them, your changes will be lost.". Instead, you should edit the python files or the *.rst files.

Added

File: docs/source/_API-Header.rst
Comment (omichel) [inline docs/source/_API-Header.rst:1]:
```suggestion
> **Note:** _Do not edit these wiki pages as they are automatically generated from the Python docstrings._
```

**Review by omichel — CHANGES_REQUESTED** [review]
Just a small layout detail (it looks better to me this way, as it stands even more out of the regular documentation text).

**Review by omichel — APPROVED** [review]
Thank you.

**Review by DavidMansolino — APPROVED** [review]

**Pull request merged**
